core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1700,10 +1700,7 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
         }
       }
 
-      petKey <- if (maxAccessLevel >= WorkspaceAccessLevels.Write)
-        samDAO.getPetServiceAccountKeyForUser(workspace.namespace, userInfo.userEmail)
-      else
-        samDAO.getDefaultPetServiceAccountKeyForUser(userInfo)
+      petKey <- samDAO.getDefaultPetServiceAccountKeyForUser(userInfo)
```

Comment:
💯 We use the project specific pet because that is the one that will be used to run workflows. I suspect there will be intermittent test failures if we don't. Checking the pet was originally introduced here to fix test issues. Although we have mostly switched to gpalloc since then.